typescript/infra/src/govern/HyperlaneAppGovernor.ts

```diff
@@ -432,4 +437,42 @@ export abstract class HyperlaneAppGovernor<
       },
     };
   }
+
+  async handleProxyAdminViolation(violation: ProxyAdminViolation) {
+    const code = await this.checker.multiProvider
+      .getProvider(violation.chain)
+      .getCode(violation.proxyAdmin.address);
+
+    let call;
+    if (code !== '0x') {
+      // admin for proxy is ProxyAdmin contract
+      call = {
+        chain: violation.chain,
+        call: {
+          to: violation.actual,
+          data: violation.proxyAdmin.interface.encodeFunctionData(
+            'changeProxyAdmin',
+            [violation.proxy.address, violation.expected],
+          ),
+          value: BigNumber.from(0),
+          description: `Change proxyAdmin of transparent proxy ${violation.proxy.address} from ${violation.actual} to ${violation.expected}`,
+        },
+      };
+    } else {
+      // admin for proxy is EOA
```

Comment:
@yorhodes - do we have proxy contracts where the admin is an EOA?

Reply:
This goes against the best practice https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/proxy/transparent/TransparentUpgradeableProxy.sol#L35-L39 which we've always followed - so imo we don't need to support this case

Reply:
I did lift this logic from the sdk 

[hyperlane-monorepo/typescript/sdk/src/deploy/HyperlaneDeployer.ts](https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/d0ce06280fab6e7a5560dc0f026a264a7cf8adab/typescript/sdk/src/deploy/HyperlaneDeployer.ts#L502)

where we still support this case

Reply:
ah gotcha, I'm happy to keep the code path fwiw